File: sdk/python/test/e2e/rest/account_test/account_fee_api_test.py
import os
import unittest

from kucoin_universal_sdk.api.client import DefaultClient
from kucoin_universal_sdk.extension.interceptor.logging import LoggingInterceptor
from kucoin_universal_sdk.generate.account.fee.model_get_basic_fee_req import GetBasicFeeReqBuilder, GetBasicFeeReq
from kucoin_universal_sdk.generate.account.fee.model_get_futures_actual_fee_req import GetFuturesActualFeeReqBuilder
from kucoin_universal_sdk.generate.account.fee.model_get_spot_actual_fee_req import GetSpotActualFeeReqBuilder
from kucoin_universal_sdk.generate.account.withdrawal.model_get_withdrawal_history_old_req import \
    GetWithdrawalHistoryOldReqBuilder
from kucoin_universal_sdk.generate.account.withdrawal.model_get_withdrawal_history_req import \
    GetWithdrawalHistoryReqBuilder
from kucoin_universal_sdk.model.client_option import ClientOptionBuilder
from kucoin_universal_sdk.model.constants import GLOBAL_API_ENDPOINT, GLOBAL_FUTURES_API_ENDPOINT, \
    GLOBAL_BROKER_API_ENDPOINT
from kucoin_universal_sdk.model.transport_option import TransportOptionBuilder


class AccountApiTest(unittest.TestCase):
    def setUp(self):
        key = os.getenv("API_KEY")
        secret = os.getenv("API_SECRET")
        passphrase = os.getenv("API_PASSPHRASE")

        # create http transport option
        http_transport_option = (
            TransportOptionBuilder()
            .set_interceptors([LoggingInterceptor()])
            .build()
        )

        # create client option
        client_option = (
            ClientOptionBuilder()
            .set_key(key)
            .set_secret(secret)
            .set_passphrase(passphrase)
            .set_spot_endpoint(GLOBAL_API_ENDPOINT)
            .set_futures_endpoint(GLOBAL_FUTURES_API_ENDPOINT)
            .set_broker_endpoint(GLOBAL_BROKER_API_ENDPOINT)
            .set_transport_option(http_transport_option)
            .build()
        )

        # create API client
        client = DefaultClient(client_option)
        kucoin_rest_api = client.rest_service()
        self.api = kucoin_rest_api.get_account_service().get_fee_api()

    def test_get_basic_fee_req(self):
        """
            get_basic_fee
            Get Basic Fee - Spot/Margin
            /api/v1/base-fee
        """

        builder = GetBasicFeeReqBuilder()
        builder.set_currency_type(GetBasicFeeReq.CurrencyTypeEnum.T_0)
        req = builder.build()
        try:
            resp = self.api.get_basic_fee(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_get_futures_actual_fee_req(self):
        """
            get_futures_actual_fee
            Get Actual Fee - Futures
            /api/v1/trade-fees
        """

        builder = GetFuturesActualFeeReqBuilder()
        builder.set_symbol("XBTUSDM")
        req = builder.build()
        try:
            resp = self.api.get_futures_actual_fee(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

    def test_get_spot_actual_fee_req(self):
        """
            get_spot_actual_fee
            Get Actual Fee - Spot/Margin
            /api/v1/trade-fees
        """

        builder = GetSpotActualFeeReqBuilder()
        builder.set_symbols("BTC-USDT")
        req = builder.build()
        try:
            resp = self.api.get_spot_actual_fee(req)
            print("code: ", resp.common_response.code)
            print("msg: ", resp.common_response.message)
            print("data: ", resp.to_json())
        except Exception as e:
            print("error: ", e)
            raise e

File: sdk/golang/tests/e2e/rest/account_test/account_fee_api_test.go
package account_test

import (
	"context"
	"encoding/json"
	"fmt"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/extension/interceptor"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/api"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/common/logger"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/generate/account/fee"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/types"
	"os"
	"testing"
)

var feeApi fee.FeeAPI

func init() {
	key := os.Getenv("API_KEY")
	secret := os.Getenv("API_SECRET")
	passphrase := os.Getenv("API_PASSPHRASE")

	defaultLogger := logger.NewDefaultLogger()
	logger.SetLogger(defaultLogger)

	httpOptionBuilder := types.NewTransportOptionBuilder().
		SetKeepAlive(true).
		AddInterceptors(interceptor.NewLoggingInterceptor(false, defaultLogger))

	option := types.NewClientOptionBuilder().
		WithKey(key).
		WithSecret(secret).
		WithPassphrase(passphrase).
		WithSpotEndpoint(types.GlobalApiEndpoint).
		WithFuturesEndpoint(types.GlobalFuturesApiEndpoint).
		WithBrokerEndpoint(types.GlobalBrokerApiEndpoint).
		WithTransportOption(httpOptionBuilder.Build()).
		Build()

	client := api.NewClient(option)

	feeApi = client.RestService().GetAccountService().GetFeeAPI()
}

func TestFeeGetBasicFeeReq(t *testing.T) {
	// GetBasicFee
	// Get Basic Fee - Spot/Margin
	// /api/v1/base-fee

	builder := fee.NewGetBasicFeeReqBuilder()
	builder.SetCurrencyType(0)
	req := builder.Build()

	resp, err := feeApi.GetBasicFee(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestFeeGetSpotActualFeeReq(t *testing.T) {
	// GetSpotActualFee
	// Get Actual Fee - Spot/Margin
	// /api/v1/trade-fees

	builder := fee.NewGetSpotActualFeeReqBuilder()
	builder.SetSymbols("BTC-USDT")
	req := builder.Build()

	resp, err := feeApi.GetSpotActualFee(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}

func TestFeeGetFuturesActualFeeReq(t *testing.T) {
	// GetFuturesActualFee
	// Get Actual Fee - Futures
	// /api/v1/trade-fees

	builder := fee.NewGetFuturesActualFeeReqBuilder()
	builder.SetSymbol("XBTUSDM")
	req := builder.Build()

	resp, err := feeApi.GetFuturesActualFee(req, context.TODO())
	if err != nil {
		panic(err)
	}
	data, err := json.Marshal(resp.ToMap())
	if err != nil {
		panic(err)
	}
	fmt.Println("code:", resp.CommonResponse.Code)
	fmt.Println("message:", resp.CommonResponse.Message)
	fmt.Println("data:", string(data))
}
